src/main/java/edu/greenblitz/tobyDetermined/subsystems/Battery.java

```diff
@@ -1,19 +1,63 @@
 package edu.greenblitz.tobyDetermined.subsystems;
 
+import edu.greenblitz.tobyDetermined.OI;
 import edu.greenblitz.tobyDetermined.RobotMap;
+import edu.greenblitz.utils.FMSUtils;
+import edu.wpi.first.math.filter.LinearFilter;
 import edu.wpi.first.wpilibj.PowerDistribution;
 import edu.wpi.first.wpilibj.RobotController;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.CommandScheduler;
 
 public class Battery extends GBSubsystem {
 	
-	private static final double minVoltage = RobotMap.General.minVoltageBattery;
+	private static final double minVoltage = RobotMap.General.MIN_VOLTAGE_BATTERY;
 	private static Battery instance;
 
+	private LinearFilter voltageFilter;
+
+	private static final int LEN_OF_AVG = 50;
+
+
+
 	private static PowerDistribution pdp = new PowerDistribution();
 
 	private Battery() {
+		voltageFilter = LinearFilter.movingAverage(LEN_OF_AVG);
+		for (int i = 0; i < LEN_OF_AVG; i++) {
+			voltageFilter.calculate(this.getCurrentVoltage());
+		}
+		SmartDashboard.putBoolean("battery low",false);
+
 	}
-	
+	@Override
+	public void periodic() {
+		SmartDashboard.putNumber("current battery voltage: ", Battery.getInstance().getCurrentVoltage());
+		SmartDashboard.putNumber("current battery usage" , Battery.getInstance().getCurrentUsage());
+
+		double currentAverageVoltage;
+
+		if(isCurrentUsageLow()) {
+			currentAverageVoltage = voltageFilter.calculate(this.getCurrentVoltage());
+
+
+			if (isVoltageLow() && !FMSUtils.isRealMatch()) {
+				SmartDashboard.putBoolean("battery low",true);
+				//add led signal for low battery
+
+				disableRobot(); //delete on competition
```

Comment:
Shouldn't be deleted if the check above works

Reply:
but yesterday we did some thinking and what if the FMS wouldn't work as planned (show training match instead of competition)? i think it will be better to be safe then sorry

Reply:
Do you talk about the official FMS in comp?  Too many things related to it properly assigning matches.
Anyway in comp branch it might be better to disable the battery limiter as a whole.